### PR TITLE
(TIN-25) Change Password

### PR DESCRIPTION
**Pull request by Kr1s4an**

### CODE REVIEW

File: src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java [hunk 1]
```diff
@@ -158,10 +159,27 @@ public void generateNewPasswordForUser(String email) throws MessagingException,
         User user = userRepository.findOneByEmail(email).orElseThrow(
                 () -> new UserDoesNotExistException("User with this email does not exist"));
 
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
         user.setPassword(PasswordGenerator.generatePassword());
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: Not necessary to set plain password value to the user, after all you will override on 166 the value, just create new variable and pass it as arg1 to emailContent.createContent().

File: src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java [hunk 2]
```diff
@@ -158,10 +159,27 @@ public void generateNewPasswordForUser(String email) throws MessagingException,
...
         String content = emailContent.createContent(user.getPassword(), "classpath:email/forgotPasswordEmail.html");
         emailSender.sendEmail(user, "Forgot Password", content);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+
+        userRepository.save(user);
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: Save the user first and then send email. And you should to refactor little bit sendEmail method to receive not a user, but email or emails. That way is more clear what the code does

File: src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java [hunk 3]
```diff
@@ -158,10 +159,27 @@ public void generateNewPasswordForUser(String email) throws MessagingException,
...
+
+    public void editUserPassword(@RequestBody ChangePasswordDto changePasswordDto) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String currentUser = authentication.getName();
+
+        User user = userRepository.findOneByEmail(currentUser).orElseThrow(() ->
+                new NotLoggedInException("You are not logged in!"));
 
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        user.setPassword(passwordEncoder.encode(changePasswordDto.getNewPassword()));
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: Think how to reuse the code, removing duplication in generateNewPasswordForUser() and editUserPassword()

Comment (Kr1s4an) [reply src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: Are you talking about the BCryptPasswordEncoder?

Comment (todornikolov1) [reply src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: In general, for the repeating code.

File: src/main/java/com/volasoftware/tinder/service/UserService.java [hunk 4]
```diff
@@ -30,4 +31,6 @@ public interface UserService {
     UserProfileDto editUserProfile(@RequestBody UserProfileDto userProfileDto);
 
     void generateNewPasswordForUser(String email) throws MessagingException, IOException;
+
+    void editUserPassword(@RequestBody ChangePasswordDto changePasswordDto);
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserService.java]:
```suggestion
    void editUserPassword(ChangePasswordDto changePasswordDto);
```

File: src/main/java/com/volasoftware/tinder/service/UserService.java [hunk 5]
```diff
@@ -30,4 +31,6 @@ public interface UserService {
     UserProfileDto editUserProfile(@RequestBody UserProfileDto userProfileDto);
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserService.java]: Please, remove `@RequestBody`

File: src/main/java/com/volasoftware/tinder/utility/PasswordEncoder.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.volasoftware.tinder.utility;
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+public class PasswordEncoder {
+    static BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+    public static String encodePassword(String password){
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/utility/PasswordEncoder.java]: Add space between methods

File: src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java [hunk 7]
```diff
@@ -119,22 +118,14 @@ public UserDetailsService userDetailsService() {
...
 
     @Override
-    public UserProfileDto editUserProfile(@RequestBody UserProfileDto userProfileDto) {
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        String currentUser = authentication.getName();
-
-        User user = userRepository.findOneByEmail(currentUser).orElseThrow(() ->
-                new NotLoggedInException("You are not logged in!"));
+    public UserProfileDto editUserProfile(UserProfileDto userProfileDto) {
+        User user = loggedUser();
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java]: please, use a verb as method name

File: src/main/java/com/volasoftware/tinder/utility/PasswordEncoder.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+public class PasswordEncoder {
+    static BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+    public static String encodePassword(String password){
+        return passwordEncoder.encode(password);
+    }
+    public static boolean equals(String rawPassword, String encodedPassword){
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/utility/PasswordEncoder.java]: Add extra space here, as well

**Review by todornikolov1 — APPROVED** [review]

**Pull request merged**